appinventor/components/src/com/google/appinventor/components/runtime/util/YailMatrix.java

```diff
@@ -0,0 +1,325 @@
+package com.google.appinventor.components.runtime.util;
+
+import com.google.appinventor.components.runtime.errors.YailRuntimeError;
+import gnu.lists.LList;
+import gnu.lists.Pair;
+import java.util.ArrayList;
+import java.util.List;
+
+public class YailMatrix {
+  private final int rows;
+  private final int cols;
+  private final double[][] data;
```

Comment:
For multidimensional arrays we'll need to rethink this. We may just need it to be a flat array and then compute the index into the underlying data structure based on the indices and dimensions.